tools/generate/src/generate-adaptor.ts

```diff
@@ -49,10 +49,21 @@ const copyAndRename = async (
       await fs.mkdir(targetPath, { recursive: true });
       await copyAndRename(sourcePath, targetPath, adaptorName);
     } else if (item.isFile()) {
-      let content = await fs.readFile(sourcePath, 'utf8');
-      content = content.replace(/{{TEMPLATE}}/g, adaptorName);
-
-      await fs.writeFile(targetPath, content);
+      const ext = path.extname(item.name).toLowerCase();
+      if (
+        ext === '.png' ||
```

Comment:
Oh good catch! That explains why the images were messed up. Nice one.

I would use a regex for this:

`if(/(png|jpe?g|gif|svg|webp)$/.test(ext)) { ... }`

The dollar means "end of string", so this is a bit like string.endsWith()`. I also snuck webp in there just for future reference.